backend/engine/plugins/lib/write_npmrc.py

```diff
@@ -79,9 +79,29 @@ def build_npm_config(scope, registry, token, username, email, **kwargs):
     )
 
 
+def build_npm_auth_token_config(scope, registry, authToken, email, **kwargs):
+    """
+    gets the variables necessary to return the npmrc config with _authToken
+    This provides npm the necessary info to get packages from private sources.
+    **kwargs exists at the end to pick up anything extra in the config dictionary that we dont need.
+    """
+    return (
+        f"@{scope}:registry=https://{registry}\n"
+        f"//{registry}:_authToken={authToken}\n"
+        f"//{registry}:email={email}\n"
+        f"//{registry}:always-auth=true\n"
```

Comment:
I'm not intimately familiar with the .npmrc file format -- do we need to escape any special characters here?

Reply:
If the input were untrusted, yes. Injection would be possible.

That said, the scopes are pulled down from AWS Secrets Manager in the same file. This should never see untrusted input (and, FWIW, the current `build_npm_config` logic does no escaping, either).